--- a/src/croupier.cr
+++ b/src/croupier.cr
@@ -468,7 +468,7 @@
     # Filesystem watcher
-    @@watcher = Inotify::Watcher.new
+    @@watcher : Inotify::Watcher | Nil = nil

     # Watch for changes in inputs.
@@ -481,8 +481,9 @@
     def watch(targets : Array(String) = [] of String)
-      if @@watcher
-        @@watcher.close
+      if current_watcher = @@watcher
+        current_watcher.close
       end
       @@watcher = Inotify::Watcher.new
       targets = tasks.keys if targets.empty?
       target_inputs = inputs(targets)

-      @@watcher.on_event do |event|
+      watcher = @@watcher.not_nil!
+      watcher.on_event do |event|
         # It's a file we care about, add it to the queue
         path = Path["#{event.path}/#{event.name}"].normalize.to_s
         Log.debug { "Detected change in #{path}" }
@@ -522,7 +523,7 @@
       target_inputs.each do |input|
         # Don't watch for changes in k/v store
         next if input.lchop?("kv://")
         if File.exists? input
-          @@watcher.not_nil!.watch input, watch_flags
+          watcher.watch input, watch_flags
         else
           # It's a file that doesn't exist. To detect it
           # being created, we watch the parent directory
           # if we are not already watching it.
           path = (Path[input].parent).to_s
-          if !@@watcher.not_nil!.watching.includes?(path)
-            @@watcher.not_nil!.watch path, watch_flags
+          if !watcher.watching.includes?(path)
+            watcher.watch path, watch_flags
           end
         end
       end